File: Python/2018/Sunflowers.py
# J4

n = int(input())
plantheights = []

for i in range(n):
    x = input().split()
    x = list(map(int, x))
    plantheights.append(x)


def check(table):
    yesno = 0
    for a in range(len(table) - 1):
        for b in range(len(table)):
            if table[a][b] < table[a + 1][b]:
                pass
            else:
                yesno = 1
    for a in range(len(table)):
        for b in range(len(table) - 1):
            if table[a][b] < table[a][b + 1]:
                pass
            else:
                yesno = 1

    if yesno == 1:
        return False
    else:
        return True


def rotate(table):
    global n
    newlist = []
    for i in range(n):
        newlist.append([])
    for a in range(n):
        for b in range(n):
            newlist[a].append(table[b][-(a + 1)])
    return newlist


while not (check(plantheights)):
    plantheights = rotate(plantheights)


for i in plantheights:
    for g in i:
        if i.index(g) != len(i) - 1:
            print(g, end=' ')
        else:
            print(g)

File: Java/2018/Sunflowers.java
import java.util.ArrayList;
import java.util.Scanner;

public class Sunflowers {
    private static int[][] plantheights;
    private static int n;

    public static boolean check(int[][] table) {
        boolean flag = true;

        for (int a = 0; a < n - 1; a++) {
            for (int b = 0; b < n; b++) {
                if (table[a][b] < table[a + 1][b]) {
                    ;
                }
                else {
                    flag = false;
                }
            }
        }

        for (int a = 0; a < n; a++) {
            for (int b = 0; b < n -1 ; b++) {
                if (table[a][b] < table[a][b + 1]) {
                    ;
                }
                else {
                    flag = false;
                }
            }
        }

        return flag;
        
    }

    public static int[][] rotate(int[][] table) {
        int[][] newmatrix = new int[n][n];
        for (int a = 0; a < n; a++) {
            for (int b = 0; b < n; b++) {
                newmatrix[a][b] = table[b][n-(a+1)];
            }
        }
        return newmatrix;
    }


    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        n = scanner.nextInt();
        plantheights = new int[n][n];

        // taking input
        for (int a = 0; a < n; a++) {
            for (int b = 0; b < n; b++) {
                plantheights[a][b] = scanner.nextInt();
            }
        }

        while (!check(plantheights)) {
            plantheights = rotate(plantheights);
        }

        for (int i = 0; i < n; i++) {
            for (int g = 0; g < n; g++) {
                if (g != n - 1) System.out.print(plantheights[i][g] + " ");
                else System.out.println(plantheights[i][g]);
            }
        }

    }
}